File: recrutement/src/main/java/com/satge/recrutement/offre/OffreRequest.java
package com.satge.recrutement.offre;

import lombok.Getter;


import java.time.LocalDate;
import java.time.LocalDateTime;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

public record OffreRequest (
        Integer id,
        @NotNull(message = "100")
        @NotEmpty(message = "100")
        String title,
        @NotNull(message = "100")
        @NotEmpty(message = "100")
        String description,
        @NotNull(message = "100")
        @NotEmpty(message = "100")
        String responsabilites,
        @NotNull(message = "100")
        @NotEmpty(message = "100")

        String profil,
        @NotNull(message = "100")
        @NotEmpty(message = "100")

        LocalDate expirationDate,
        Boolean shareable,
        TypeContrat typeContrat,

        String publisher

){


}

File: front/src/app/services/models/offre-request.ts
/* tslint:disable */
/* eslint-disable */
export interface OffreRequest {
  description: string;
  expirationDate: string;
  id?: number;
  profil: string;
  publisher?: string;
  responsabilites: string;
  shareable?: boolean;
  title: string;
  typeContrat?: 'Cdi' | 'Stage' | 'cdd';
}
